camp/asher-miniproject/src/main/java/frc/robot/subsystems/MotorSubsystem.java

```diff
@@ -0,0 +1,129 @@
+// ************************************************************
+// Bishop Blanchet Robotics
+// Home of the Cybears
+// FRC - First Age - 2026
+// File: MotorSubsystem.java
+// ************************************************************
+
+// ʕ •ᴥ•ʔ ʕ•ᴥ•  ʔ ʕ  •ᴥ•ʔ ʕ •`ᴥ´•ʔ ʕ° •° ʔ ʕ •ᴥ•ʔ ʕ•ᴥ•  ʔ ʕ  •ᴥ•ʔ ʕ •`ᴥ´•ʔ ʕ° •° ʔ 
+
+package frc.robot.subsystems;
+import com.ctre.phoenix6.hardware.CANcoder;
+import com.ctre.phoenix6.hardware.TalonFX;
+import com.ctre.phoenix6.StatusCode;
+import com.ctre.phoenix6.configs.CANcoderConfiguration;
+import com.ctre.phoenix6.configs.TalonFXConfiguration;
+import com.ctre.phoenix6.controls.DutyCycleOut;
+import com.ctre.phoenix6.signals.InvertedValue;
+import com.ctre.phoenix6.signals.NeutralModeValue;
+
+import edu.wpi.first.wpilibj.DataLogManager;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+import static java.lang.Math.max;
+import static java.lang.Math.min;
+
+import edu.wpi.first.math.MathUtil.*;
+
+import frc.robot.Constants;
+
+/**
+ * The MotorSubsystem is responsible for controlling a motor and its associated encoder.
+ * It provides methods to set motor speed, retrieve the position of the encoder, and configure
+ * the motor and encoder settings.
+ */
+public class MotorSubsystem extends SubsystemBase {
+    private final TalonFX motor;
+    private final DutyCycleOut eeDutyCycle = new DutyCycleOut(0.0);
+    private final CANcoder motorEncoder;
+
+    /**
+     * Constructs a MotorSubsystem with a motor and an encoder.
+     *
+     * @param motorCan The CAN ID of the motor controller (TalonFX).
+     * @param encoderCan The CAN ID of the encoder (CANcoder).
+     */
+    public MotorSubsystem(int motorCan, int encoderCan) {
+        this.motor = new TalonFX(motorCan);
+        configureMotor();
+
+        this.motorEncoder = new CANcoder(encoderCan);
+        configureAngleEncoder();
+    }
+
+    /**
+     * Retrieves the position of the endodger in degrees.
```

Comment:
what's an endodger? Maybe you mean encoder?

Reply:
typo fixed